text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,276 @@
+1. Reads the following information from the image config (if any of these are missing, the command fails):
+    - `id` (must include a `namespace` and `name` in the form `<namespace>/<name>`)
+    - `version`
+1. Opens a link to a new Github Issue in the user's browser
+    - The user must be logged into Github
+    - The link will use [Github query parameters](https://help.github.com/en/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters) to pre-populate the title, body, and labels.
+    - The issue will be opened against the `https://github.com/buildpacks/registry-index` repo.
+    - The issue body will contain structured data that defines the buildpack id, version, digest, and url.
+1. A Github action will detect the new issue and do the following:
+    - If this is the first time the `namespace` is used, it will add the namespace to the `buildpacks/registry-owners` repo with the Github user who opened the issue as the owner. The `buildpacks/registry-owners` repo defines the [namespace ownership as described in RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md#namespace-ownership).
+    - If this is *not* the first time the `namespace` is used, it will confirm that the Github user who opened the issue is an owner of the buildpack.
+    - Create a commit against the master branch of the `https://github.com/buildpacks/registry-index` repo using a Gitub token (i.e. all commits in that repo will be made by the same "user"). The commit will add the buildpack version described in the issue.
+    - Close the Github issue.
+
+The following behavior will execute when the configured registry is of type `git`.
+
+1. Pulls the image manifest for the buildpackage at `url` if it does not already exist locally.
+1. Reads the following information from the image manifest (if any of these are missing, the command fails):
+    - `id` (must include a `namespace` and `name` in the form `<namespace>/<name>`)
+    - `version`
+1. Creates a `local` branch on the local registry index cache
+1. Makes a Git commit to the local registry index with a JSON payload represeting the buildpack version as defined in [RFC-0022](https://github.com/buildpacks/rfcs/blob/master/text/0022-client-side-buildpack-registry.md)
+1. Uses Git to push the commit to the registry index's `origin`.
+1. Updates the `master` branch of the local registry cache from `origin`.
+
+Options:
+
+* `--buildpack-registry, -R` - the id of a registry configured in `.pack/config.toml`
+* `--auth <token>` - If the registry type would default to opening a web browser, it will instead perform the action using an API. In the case of Github, it will use the provided token to invoke the [Issues API](https://developer.github.com/v3/issues/)
+
+**NOTE:** Github Issues that are not recognized as a request to add/yank a buildpack version will be automatically closed. All PRs will also be automatically closed.
```

Comment:
If we choose to use PRs instead, we can just close the Issues tab.

Reply:
FWIW, IIRC, PR's weren't possible due to the lack of API support from GH for creating PRs from patches.